pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -102,139 +98,41 @@ func NewController(name, operatorNamespace, targetNamespace, targetOperandVersio
 		podsLister:                   podLister,
 		delegate:                     delegate,
 		openshiftClusterConfigClient: openshiftClusterConfigClient,
-		eventRecorder:                eventRecorder.WithComponentSuffix("workload-controller"),
 		versionRecorder:              versionRecorder,
 		queue:                        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), name),
 	}
 
-	return controllerRef
+	c := factory.New()
+	for _, nsi := range namespacedInformers {
+		c.WithNamespaceInformer(nsi, targetNamespace)
```

Comment:
side note: the interface is kind of broken because we already specify the namespace at creation time, for example `kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().ConfigMaps().Informer()`

Reply:
`factory.Informer` does not in general enforces the informer to be namespaced I believe, so if we were to reuse this controller outside of the context of the APIServerControllerSet, this would still be a valid use